Breakout parsers

tobsan opened the pull request:
Breaking out parsers for most gateways into their own classes, which can be properly unit tested.

Commit:
put gateway classes and files in own dirs

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
Breaks out the cgroupsgateway parser to own class.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
breaks out the device node gateway parser to its own class

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
breaks out the environment gateway parser into its own class

This also makes the old unit-tests for the envgateway completely
obsolete, so they have been removed.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
breaks out file gateway parser to its own class

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
breaks out the dbus gateway parser into own class

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
creates nicer cmake structure for gateways

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
breaks out jobs in CMake to own dir

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
m_state is now private to gateway class

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

breakreturn commented on libsoftwarecontainer/include/gateway.h:
```diff
@@ -154,11 +154,13 @@ class Gateway
     static constexpr const char *ENABLED_FIELD = "enabled";
     static constexpr const char *XDG_RUNTIME_DIR_VARIABLE_NAME = "XDG_RUNTIME_DIR";
```
While at it, could you also see if the defines make sense to have in the base class?

tobsan replied:
Done, it didn't.

breakreturn commented on libsoftwarecontainer/unit-test/dbusgatewayparser_unittest.cpp:
```diff
@@ -0,0 +1,127 @@
+    
+    ASSERT_NE(json_array_size(sessionConf), empty);
+    ASSERT_NE(json_array_size(systemConf), empty);
+}
```
Please add descriptive comments to all tests. Everywhere ;)

tobsan replied:
Done!

Commit:
Some constants move to waylandgateway

They were only used there, but still defined in the gateway class. Not
any more.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>
